cmd/noms/diff/diff_test.go

```diff
@@ -237,85 +342,131 @@ func TestNomsListDiff(t *testing.T) {
+
 		buf = &bytes.Buffer{}
-		Diff(buf, l3, l4, leftRight)
+		PrintDiff(buf, l3, l4, leftRight)
 		assert.Equal(expected2, buf.String())
 
-		buf = &bytes.Buffer{}
-		Diff(buf, l5, l6, leftRight)
+		paths = pathsFromDiff(l3, l4, leftRight)
+		assert.Equal(expectedPaths2, paths)
 
+		buf = &bytes.Buffer{}
+		PrintDiff(buf, l5, l6, leftRight)
 		assert.Equal(expected3, buf.String())
+
+		paths = pathsFromDiff(l5, l6, leftRight)
+		assert.Equal(expectedPaths3, paths)
 	}
 
 	tf(true)
 	tf(false)
 }
 
-func TestNomsBlobDiff(t *testing.T) {
+func TestNomsDiffPrintBlob(t *testing.T) {
 	assert := assert.New(t)
 
 	expected := "-   Blob (2.0 kB)\n+   Blob (11 B)\n"
+	expectedPaths1 := []string{``}
 	b1 := types.NewBlob(strings.NewReader(strings.Repeat("x", 2*1024)))
 	b2 := types.NewBlob(strings.NewReader("Hello World"))
 
 	tf := func(leftRight bool) {
 		buf := &bytes.Buffer{}
-		Diff(buf, b1, b2, leftRight)
+		PrintDiff(buf, b1, b2, leftRight)
 		assert.Equal(expected, buf.String())
+
+		paths := pathsFromDiff(b1, b2, leftRight)
+		assert.Equal(expectedPaths1, paths)
 	}
 
 	tf(true)
 	tf(false)
 }
 
-func TestNomsTypeDiff(t *testing.T) {
+func TestNomsDiffPrintType(t *testing.T) {
 	assert := assert.New(t)
 
 	expected1 := "-   List<Number>\n+   List<String>\n"
+	expectedPaths1 := []string{``}
 	t1 := types.MakeListType(types.NumberType)
 	t2 := types.MakeListType(types.StringType)
 
 	expected2 := "-   List<Number>\n+   Set<String>\n"
+	expectedPaths2 := []string{``}
 	t3 := types.MakeListType(types.NumberType)
 	t4 := types.MakeSetType(types.StringType)
 
 	tf := func(leftRight bool) {
 		buf := &bytes.Buffer{}
-		Diff(buf, t1, t2, leftRight)
+		PrintDiff(buf, t1, t2, leftRight)
 		assert.Equal(expected1, buf.String())
 
+		paths := pathsFromDiff(t1, t2, leftRight)
+		assert.Equal(expectedPaths1, paths)
+
 		buf = &bytes.Buffer{}
-		Diff(buf, t3, t4, leftRight)
+		PrintDiff(buf, t3, t4, leftRight)
 		assert.Equal(expected2, buf.String())
+
+		paths = pathsFromDiff(t3, t4, leftRight)
+		assert.Equal(expectedPaths2, paths)
 	}
 
 	tf(true)
 	tf(false)
 }
 
-func TestNomsRefDiff(t *testing.T) {
+func TestNomsDiffPrintRef(t *testing.T) {
+	assert := assert.New(t)
+
 	expected := "-   fckcbt7nk5jl4arco2dk7r9nj7abb6ci\n+   i7d3u5gekm48ot419t2cot6cnl7ltcah\n"
+	expectedPaths1 := []string{``}
 	l1 := createList(1)
 	l2 := createList(2)
 	r1 := types.NewRef(l1)
 	r2 := types.NewRef(l2)
 
 	tf := func(leftRight bool) {
 		buf := &bytes.Buffer{}
-		Diff(buf, r1, r2, leftRight)
-
+		PrintDiff(buf, r1, r2, leftRight)
 		test.EqualsIgnoreHashes(t, expected, buf.String())
+
+		paths := pathsFromDiff(r1, r2, leftRight)
+		assert.Equal(expectedPaths1, paths)
 	}
 
 	tf(true)
 	tf(false)
 }
+
+func TestCopyPath(t *testing.T) {
+	assert := assert.New(t)
+
+	testCases := []string{
+		``,
+		`["key"]`,
+		`["key"].field1`,
+		`["key"]@key.field1`,
+	}
+
+	for _, s1 := range testCases {
+		p := mustParsePath(assert, s1)
+		p1 := copyPath(p)
```

Comment:
You should really also modify `p` after this to make sure it's an actual copy. Like, `p[0] = <something>`